src/domain/accounts/services/src/use_cases/create_account_use_case_impl.rs

```diff
@@ -82,30 +119,79 @@ impl CreateAccountUseCaseImpl {
 impl CreateAccountUseCase for CreateAccountUseCaseImpl {
     async fn execute(
         &self,
-        creator_account: &Account,
+        creator_account: Option<&Account>,
         account_name: &odf::AccountName,
-        options: CreateAccountUseCaseOptions,
+        predefined_fields: PredefinedAccountFields,
     ) -> Result<Account, CreateAccountError> {
-        let email = if let Some(email) = options.email {
+        let email = if let Some(email) = predefined_fields.email {
             email
         } else {
             Self::generate_email(creator_account, account_name)?
         };
```

Comment:
Prefer a separate entry point for "derived email" instead of `Option<&Account>`. You could have normal `execute` and a `execute_derived` taking a creator account, which internally may share most of the algorithm via private methods.

Reply:
Same story for notification - notify in "execute_derived" only without `if` statement.